sdk/identity/Azure.Identity/samples/ClientSideUserAuthentication.md

```diff
@@ -0,0 +1,105 @@
+
+The second difference is here the application is preventing the credential from automatically initiating user interaction. Even though the application authenticates the user before the credential is used, further interaction might still be needed, for instance in the case that the user's refresh token expires, or a specific method require additional consent or authentication. 
+
+By setting the option `DisableAutomaticAuthentication` to `true` the credential will fail to automatically authenticate calls where user interaction is necessary. Instead, the credential will throw an `AuthenticationRequiredException`. The following example demonstrates an application handling such an exception to prompt the user to authenticate only after some application logic has completed.
+
+```C# Snippet:Identity_ClientSideUserAuthentication_DisableAutomaticAuthentication_ExHandling
+try
+{
+    client.GetSecret("secret");
+}
+catch (AuthenticationRequiredException e)
+{
+    await EnsureAnimationCompleteAsync();
+
+    await credential.AuthenticateAsync(e.TokenRequestContext);
+
+    client.GetSecret("secret");
+}
+```
+
+## Persisting user authentication data
+
+Quite often applications desire the ability to be run multiple times without having to reauthenticate the user on each execution. This requires that data from the original authentication be persisted outside of the application memory, so that it can authenticate silently on subsequent executions. Specifically two pieces of data need to be persisted, the `TokenCache` and the `AuthenticationRecord`. 
+
+### Persisting the TokenCache
+
+The `TokenCache` contains all the data needed to silently authenticate, one or many accounts. It contains sensitive data such as refresh tokens, and access tokens and must be protected to prevent compromising the accounts it houses tokens for. The `Azure.Identity` library provides the `PersistentTokenCache` class which by default will protect and persist the cache using available platform data protection. 
+
+To use the `PersistentTokenCache` to persist the cache of any credential simply set the `TokenCache` option.
+
+```C# Snippet:Identity_ClientSideUserAuthentication_Persist_TokenCache
+var credential = new InteractiveBrowserCredential(new InteractiveBrowserCredentialOptions { TokenCache = new PersistentTokenCache() });
+```
+
+### Persisting the AuthenticationRecord
+
+The `AuthenticationRecord` which is returned from the `Authenticate` and `AuthenticateAsync`, contains data identifying an authenticated account. It is needed to identify the appropriate entry in the `TokenCache` to silently on subsequent executions. There is no sensitive data in the `AuthenticationRecord` so it can be persisted in a non-protected state.
```

Comment:
" to silently on subsequent executions" ?

Reply:
fixed